foyer/validator.py

```diff
@@ -193,5 +193,5 @@ def validate_overrides(self):
 
 if __name__ == '__main__':
     from foyer.tests.utils import get_fn
-    v = Validator(get_fn('validate_types.xml'))
-    v = Validator(get_fn('validationerror_validate_types.xml'))
+    Validator(get_fn('validate_types.xml'))
+    Validator(get_fn('validationerror_validate_types.xml'))
```

Comment:
Is there a reason we made `Validator` a class? It seems like it's used just like a function with nested functions.

Reply:
That decision was made before my time so I can't speak to other ideas that were floated around. However, I agree it could probably be gutted in favor of its constituent functions but I'm not sure there's a downside to having it as a class.